diff --git a/Makefile b/Makefile
index bf5ae42..6f472f0 100644
--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,8 @@
 # First, specify your toolchain to compile the library.
 # Libraries available are compiled with riscv32-unknown-elf-
 
-RV_PREFIX = 	riscv32-unknown-elf-
+XLEN ?= 64
+RV_PREFIX = 	riscv${XLEN}-unknown-elf-
 CC = 			$(RV_PREFIX)gcc
 AR = 			$(RV_PREFIX)ar
 OBJDUMP = 		$(RV_PREFIX)objdump
@@ -14,7 +15,7 @@ M_EXTENSION 	?= Y
 C_EXTENSION 	?= N
 F_EXTENSION		?= N
 
-ARCH = rv32i
+ARCH = rv${XLEN}i
 
 DEBUG 			?= Y
 
@@ -33,13 +34,14 @@ endif
 ARCH := $(addsuffix _zicsr_zifencei,$(ARCH))
 
 # Compiler flags
-CFLAGS = 		-march=$(ARCH) -mabi=ilp32
+# CFLAGS = 		-march=$(ARCH) -mabi=ilp32
+CFLAGS = 		-march=$(ARCH) -mabi=lp64
 CFLAGS +=		-Wall -Werror -Wno-unused-but-set-variable
 CFLAGS +=		-O2
 CFLAGS += 		-c
 
 ifeq ($(DEBUG), Y)
-CFLAGS +=		-g
+# CFLAGS +=		-g
 endif
 
 # Include
